tests/test_server.py

```diff
@@ -2027,3 +2027,27 @@ async def test_set_param_v2(server, client, environment):
     assert res.code == 200
     parameters = res.result["parameters"]
     assert len(parameters) == 2
+
+
+async def test_delete_active_version(client, clienthelper, server, environment):
+    """
+    Test that the active version cannot be deleted
+    """
+    version = await clienthelper.get_version()
+    assert version == 1
+    res1 = "test::Resource[agent1,key=key1]"
+    res2 = "test::Resource[agent1,key=key2]"
+    resources = [
+        {"key": "key1", "value": "value", "id": f"{res1},v={version}", "requires": [], "purged": False, "send_event": False},
+        {"key": "key2", "value": "value", "id": f"{res2},v={version}", "requires": [], "purged": False, "send_event": False},
+    ]
+
+    await clienthelper.put_version_simple(resources, version)
+
+    result = await client.release_version(environment, version, True, const.AgentTriggerMethod.push_full_deploy)
```

Comment:
I also don't see why we need `push=True` here?
```suggestion
    result = await client.release_version(environment, version, push=True, const.AgentTriggerMethod.push_full_deploy)
```